process/smartContract/process.go

```diff
@@ -920,6 +921,9 @@ func (sc *scProcessor) createSCRForSender(
 	acntSnd state.UserAccountHandler,
 	callType vmcommon.CallType,
 ) (*smartContractResult.SmartContractResult, *big.Int) {
+	if gasRefund == nil {
```

Comment:
Perhaps we can create a function in core, `core.CoalesceBigInt()` that does exactly this (and refactor in some other places to use it).